File: LC-7/LC-7.cpp
#include <climits>

using namespace std;

class Solution {
public:
    /**
     * Concepts: Math Approach
     */ 
    int reverse(int x) {
        if (x == 0)
            return 0;
        long long rev = 0;
        while (x) {
            rev = rev * 10 + x % 10;
            x /= 10;
            if (rev > INT_MAX || rev < INT_MIN)
                return 0;
        }
        return (int)rev;
    }
};

File: LC-7/LC-7.js
/**
 * @param {number} x
 * @return {number}
 */

const INT_MAX = 2147483647,
  INT_MIN = -2147483647 - 1

var reverse = function (x) {
  /**
   * Concepts: Math Approach
   */

  if (x === 0) {
    return 0
  }

  let rev = 0
  while (x) {
    rev = rev * 10 + parseInt(x % 10)
    x = parseInt(x / 10)
    if (rev > INT_MAX || rev < INT_MIN) {
      return 0
    }
  }
  return parseInt(rev)
}
